### PR TITLE
Fix more definition and inference bugs

### PR DESCRIPTION
**Pull request by jmbockhorst**
Fix some bugs due to type vs union constructor definitions, if else expr parsing, and fix/add/enable more tests.

### CODE REVIEW

File: test/definitionProviderTests/importDefinition.test.ts [hunk 1]
```diff
@@ -113,6 +114,7 @@ import App exposing (Page)
 --@ App.elm
 module App exposing (Page)
 type Page = Home
+--X
```
Comment (Razzeee) [inline test/definitionProviderTests/importDefinition.test.ts:117]: Are these changes necessary? So far the code only checks if we changed to the correct file I think, are you planning to change this?

File: test/definitionProviderTests/qualifiedImportDefinition.test.ts [hunk 2]
```diff
@@ -105,6 +105,44 @@ update msg model = (model, Ports.foo "blah")
 --@ Ports.elm
 port module Ports exposing (foo)
 port foo : String -> Cmd msg
+`;
+    await testBase.testDefinition(source);
+  });
+
+  xit(`test union constructor should not resolve to type declaration from other file`, async () => {
```
Comment (Razzeee) [inline test/definitionProviderTests/qualifiedImportDefinition.test.ts:112]: Still deactivated?

File: test/definitionProviderTests/importDefinition.test.ts (hunk 1 above)
Comment (jmbockhorst) [reply test/definitionProviderTests/importDefinition.test.ts:117]: I had added the ability to check the node position too in https://github.com/elm-tooling/elm-language-server/commit/f63617aa42fb0b40519187cfc065f6d5a1416bc4#.

File: test/definitionProviderTests/qualifiedImportDefinition.test.ts (hunk 2 above)
Comment (jmbockhorst) [reply test/definitionProviderTests/qualifiedImportDefinition.test.ts:112]: Yeah, for some reason it wouldn't select the right node. It resolves to `User { data : String }` starting at character 12, which is correct. But the `--X` was selecting a node starting at character 10. Not sure why

Comment (jmbockhorst) [reply test/definitionProviderTests/qualifiedImportDefinition.test.ts:112]: Looks like I didn't hook it up right.

File: test/definitionProviderTests/importDefinition.test.ts (hunk 1 above)
Comment (Razzeee) [reply test/definitionProviderTests/importDefinition.test.ts:117]: Right, forgot about that :)
